fastjet/internal/delaunay/triangle_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"testing"
+)
+
+func TestTriangleEquals(t *testing.T) {
+	tests := []struct {
+		a, b *Triangle
+		want bool
+	}{
+		{NewTriangle(NewPoint(3, 2), NewPoint(0, 0), NewPoint(5, 0)), NewTriangle(NewPoint(3, 2), NewPoint(0, 0), NewPoint(5, 0)), true},
+		{NewTriangle(NewPoint(3, 2), NewPoint(0, 0), NewPoint(5, 0)), NewTriangle(NewPoint(5, 0), NewPoint(3, 2), NewPoint(0, 0)), true},
+		{NewTriangle(NewPoint(3, 2), NewPoint(0, 0), NewPoint(5, 0)), NewTriangle(NewPoint(3, 1), NewPoint(0, 0), NewPoint(5, 0)), false},
+		{NewTriangle(NewPoint(3, 2), NewPoint(0, 0), NewPoint(5, 0)), NewTriangle(NewPoint(5, 0), NewPoint(3, 1), NewPoint(0, 0)), false},
+	}
+	for _, test := range tests {
+		got := test.a.Equals(test.b)
+		if got != test.want {
+			t.Fatalf("%v == %v,\n got = %v, want = %v", test.a, test.b, got, test.want)
+		}
+	}
+}
+
+func TestTrianglesRemove(t *testing.T) {
+	tests := []struct {
+		triangles triangles
+		toRemove  []*Triangle
+	}{
+		{
+			triangles{NewTriangle(NewPoint(0, 0), NewPoint(0, 1), NewPoint(1, 0)), NewTriangle(NewPoint(0, 0), NewPoint(5, 1), NewPoint(1, 0)), NewTriangle(NewPoint(0, 5), NewPoint(0, 1), NewPoint(1, 0))},
+			[]*Triangle{NewTriangle(NewPoint(0, 0), NewPoint(0, 1), NewPoint(1, 0)), NewTriangle(NewPoint(0, 5), NewPoint(0, 1), NewPoint(1, 0))},
+		},
+	}
+	for _, test := range tests {
+		triangles := test.triangles.remove(test.toRemove[:]...)
```

Comment:
this is a bit of a tricky and finicky line.
you rely on the fact that a new slice will be created when using the ellipsis operator.

it would be much clearer (and robust) if you rewrite it like so:

```go
toRemove := make([]*Triangle, len(test.toRemove))
copy(toRemove, test.toRemove)
triangles := test.triangles.remove(toRemove...)
for _, tri := range triangles {
    for _, removed := range test.toRemove {
      // ...
```